base/applications/osk/settings.c

```diff
@@ -37,133 +31,49 @@ BOOL LoadDataFromRegistry(VOID)
-        RegCloseKey(hKey);
-        return FALSE;
-    }
-
-    /* Load the dialog layout value */
-    Globals.bIsEnhancedKeyboard = (dwValue != 0);
-
-    /* Query the key */
-    lResult = RegQueryValueExW(hKey,
-                               L"ClickSound",
-                               0,
-                               0,
-                               (BYTE *)&dwValue,
-                               &cbData);
-
-    if (lResult != ERROR_SUCCESS || cbData != sizeof(dwValue))
-    {
-        /* Bail out and return FALSE if we fail */
-        RegCloseKey(hKey);
-        return FALSE;
+        /* Bail out */
+        DPRINT("LoadDataFromRegistry(): Failed to open the application's key! (Error - %li)\n", lResult);
+        return lResult;
     }
 
-    /* Load the sound on click value event */
-    Globals.bSoundClick = (dwValue != 0);
-
-    /* Query the key */
+    /* Load the specific value based on the parameter caller, lpValueDataName */
     lResult = RegQueryValueExW(hKey,
-                               L"WindowLeft",
+                               lpValueDataName,
                                0,
                                0,
                                (BYTE *)&dwValue,
                                &cbData);
 
-    if (lResult != ERROR_SUCCESS || cbData != sizeof(dwValue))
+    if (lResult != ERROR_SUCCESS)
     {
-        /* Bail out and return FALSE if we fail */
-        RegCloseKey(hKey);
-        return FALSE;
-    }
 
-    /* Load the X value data of the dialog's coordinate */
-    Globals.PosX = dwValue;
-
-    lResult = RegQueryValueExW(hKey,
-                               L"WindowTop",
-                               0,
-                               0,
-                               (BYTE *)&dwValue,
-                               &cbData);
-
-    if (lResult != ERROR_SUCCESS || cbData != sizeof(dwValue))
-    {
-        /* Bail out and return FALSE if we fail */
+        /* Bail out */
+        DPRINT("LoadDataFromRegistry(): Failed to load the following value - \"%S\". (Error - %li)\n", lpValueDataName, lResult);
         RegCloseKey(hKey);
-        return FALSE;
+        return lResult;
     }
 
-    /* Load the Y value data of the dialog's coordinate */
-    Globals.PosY = dwValue;
-
-    lResult = RegQueryValueExW(hKey,
-                               L"AlwaysOnTop",
-                               0,
-                               0,
-                               (BYTE *)&dwValue,
-                               &cbData);
-
-    if (lResult != ERROR_SUCCESS || cbData != sizeof(dwValue))
+    /* Is the buffer's size too small to query the required data? */
+    if (cbData != sizeof(dwValue))
     {
-        /* Bail out and return FALSE if we fail */
+        /* It is therefore bail out */
+        DPRINT("LoadDataFromRegistry(): The buffer is too small to hold the data!\n");
         RegCloseKey(hKey);
-        return FALSE;
+        return lResult;
```

Comment:
`lResult` is ERROR_SUCCESS here, so you'll want to return an explicit error code instead.